texar/modules/pretrained/xlnet_model_utils.py

```diff
@@ -0,0 +1,342 @@
+# Copyright 2019 The Texar Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Model Utils of XLNet Modules.
+
+Adapted from
+https://github.com/zihangdai/xlnet/blob/master/modeling.py
+"""
+
+from typing import Any, Dict, Optional, Tuple
+
+import torch
+import torch.nn as nn
+import torch.nn.functional as F
+
+from texar.core import get_layer
+from texar.module_base import ModuleBase
+
+__all__ = [
+    "PositionWiseFF",
+    "RelativeMultiheadAttention",
+    "RelativePositionalEncoding",
+]
+
+
+class PositionWiseFF(ModuleBase):
```

Comment:
Is it the same as the position_wise networks used in transformer models? https://github.com/gpengzhi/texar-pytorch/blob/xlnet/texar/modules/encoders/transformer_encoder.py#L36

If so, let's use the same way to construct this part. I'd prefer the previous way that uses texar FeedForwardNet